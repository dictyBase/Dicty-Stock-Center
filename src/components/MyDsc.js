// @flow
import React from "react"
import { connect } from "react-redux"
import PanelWrapper from "components/common/PanelWrapper"
import { Link } from "react-router-dom"
import { withStyles } from "@material-ui/core/styles"
import Grid from "@material-ui/core/Grid"
import type { MapStateToProps } from "react-redux"

const styles = theme => ({
  container: {
    paddingRight: "15px",
    paddingLeft: "15px",
    marginRight: "auto",
    marginLeft: "auto",
    width: "75%",
    "@media (min-width: 1300px)": {
      width: "1270px",
    },
  },
  breadcrumb: {
    padding: "8px 15px",
    marginBottom: "20px",
    listStyle: "none",
    backgroundColor: "#f5f5f5",
    borderRadius: "4px",
    "&:active": {
      color: "#777",
    },
  },
  breadcrumbFirstItem: {
    display: "inline-block",
  },
  breadcrumbItem: {
    display: "inline-block",
    "&:before": {
      padding: "0 5px",
      color: "#ccc",
      content: "'/'",
    },
  },
  header: {
    marginTop: "25px",
    marginBottom: "25px",
    borderBottom: "1px solid #eee",
    textAlign: "center",
  },
  innerPanel: {
    padding: "10px",
  },
})

type Props = {
  /** the object that contains auth data from current state */
  auth: Object,
  /** Material-UI styling */
  classes: Object,
}

/**
 * Personalized component that displays the user's login information (ID, email, name, provider)
 */

export const MyDsc = (props: Props) => {
  const {
    auth: {
      user: { data },
      provider,
    },
    classes,
  } = props

  return (
    <Grid container justify="center" className={classes.container}>
      <Grid item xs={12}>
        <ol className={classes.breadcrumb}>
          <li className={classes.breadcrumbFirstItem}>
            <Link to="/">Home</Link>
          </li>
          <li className={classes.breadcrumbItem}>My DSC</li>
        </ol>
      </Grid>
      <Grid item xs={12}>
        <div className={classes.header}>
          <h2>My DSC</h2>
        </div>
      </Grid>
      <Grid item xs={8}>
        <PanelWrapper title="Personal Information">
          <div className={classes.innerPanel}>
            {data.id && <h3>Id: {data.id}</h3>}
            {data.attributes.email && <h3>Email: {data.attributes.email}</h3>}
            {data.attributes.name && <h3>Name: {data.attributes.name}</h3>}
            <h3>Provider: {provider}</h3>
          </div>
        </PanelWrapper>
      </Grid>
    </Grid>
  )
}

const mapStateToProps: MapStateToProps<*, *, *> = ({ auth }) => ({ auth })

export default connect(mapStateToProps)(withStyles(styles)(MyDsc))
