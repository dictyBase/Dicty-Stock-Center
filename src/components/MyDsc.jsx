import React, { Component } from 'react'
import { Panel, PanelBody } from 'dicty-components-panel'
import { Link } from 'react-router-dom'
import { Flex, Box } from 'rebass'
import { Container, DictyHeader, Breadcrumb } from 'styles'
import FontAwesome from 'react-fontawesome'

export default class MyDsc extends Component {
    displayName = 'user profile';
    render() {
        const { user, provider } = this.props.auth
        return (
            <Container>
                <Flex wrap justify="center">
                    <Box w={ 1 }>
                        <Breadcrumb>
                          <li><Link to="/">Home</Link></li>
                          <li>My DSC</li>
                        </Breadcrumb>
                    </Box>
                    <Box w={ 1 }>
                        <DictyHeader>
                            <h2>My DSC</h2>
                        </DictyHeader>
                    </Box>
                    <Box w={ 3 / 4 }>
                        <Panel collapse>
                            <PanelBody>
                                <h1>
                                    <FontAwesome name="user" /> Personal Information
                                </h1>
                                <hr />
                                <h3>Name: { user.name }</h3>
                                <h3>Email: { user.email }</h3>
                                <h3>Provider: { provider }</h3>
                            </PanelBody>
                        </Panel>
                    </Box>
                </Flex>
            </Container>
        )
    }
}

