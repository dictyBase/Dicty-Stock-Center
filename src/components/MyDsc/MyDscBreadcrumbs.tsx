import React from "react"
import { Link } from "react-router-dom"
import useStyles from "./myDscStyles"

/**
 * MyDscBreadcrumbs displays the breadcrumbs at the top of the MyDsc page.
 */

const MyDscBreadcrumbs = () => {
  const classes = useStyles()

  return (
    <ol className={classes.breadcrumb}>
      <li className={classes.breadcrumbFirstItem}>
        <Link to="/">Home</Link>
      </li>
      <li className={classes.breadcrumbItem}>MyDSC</li>
    </ol>
  )
}

export default MyDscBreadcrumbs
