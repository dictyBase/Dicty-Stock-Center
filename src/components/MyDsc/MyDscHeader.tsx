import React from "react"
import useStyles from "./myDscStyles"

/**
 * MyDscHeader displays the header on the MyDsc page.
 */

const MyDscHeader = () => {
  const classes = useStyles()

  return (
    <div className={classes.header}>
      <h2>My DSC</h2>
    </div>
  )
}

export default MyDscHeader
