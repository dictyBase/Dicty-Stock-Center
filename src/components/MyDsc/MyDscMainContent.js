// @flow
import React from "react"
import PanelWrapper from "components/common/PanelWrapper"
import useStyles from "./myDscStyles"

type Props = {
  /** User account data */
  data: {
    id: string,
    attributes: {
      email: string,
      name: string,
    },
  },
  /** Account provider (i.e. orcid, google) */
  provider: string,
}

/**
 * MyDscMainContent currently displays the user's account information.
 */

const MyDscMainContent = ({ data, provider }: Props) => {
  const classes = useStyles()

  return (
    <PanelWrapper title="Personal Information">
      <div className={classes.innerPanel}>
        {data.id && <h3>Id: {data.id}</h3>}
        {data.attributes.email && <h3>Email: {data.attributes.email}</h3>}
        {data.attributes.name && <h3>Name: {data.attributes.name}</h3>}
        <h3>Provider: {provider}</h3>
      </div>
    </PanelWrapper>
  )
}

export default MyDscMainContent
