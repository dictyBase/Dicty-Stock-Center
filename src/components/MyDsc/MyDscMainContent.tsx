import React from "react"
import PanelWrapper from "components/common/PanelWrapper"
import useStyles from "./myDscStyles"

type Props = {
  data: {
    id: string
    email: string
    first_name: string
    last_name: string
  }
  provider: string
}

/**
 * MyDscMainContent currently displays the user's account information.
 */

const MyDscMainContent = ({ data, provider }: Props) => {
  const classes = useStyles()
  const name = `${data.first_name} ${data.last_name}`

  return (
    <PanelWrapper title="Personal Information">
      <div className={classes.innerPanel}>
        <h3>Id: {data.id}</h3>
        <h3>Email: {data.email}</h3>
        <h3>Name: {name}</h3>
        <h3>Provider: {provider}</h3>
      </div>
    </PanelWrapper>
  )
}

export default MyDscMainContent
