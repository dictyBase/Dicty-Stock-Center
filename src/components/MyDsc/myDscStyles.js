import { makeStyles } from "@material-ui/core/styles"

const useStyles = makeStyles({
  container: {
    paddingRight: "15px",
    paddingLeft: "15px",
    marginRight: "auto",
    marginLeft: "auto",
    width: "75%",
    "@media (min-width: 1300px)": {
      width: "1270px",
    },
  },
  breadcrumb: {
    padding: "8px 15px",
    marginBottom: "20px",
    listStyle: "none",
    backgroundColor: "#f5f5f5",
    borderRadius: "4px",
    "&:active": {
      color: "#777",
    },
  },
  breadcrumbFirstItem: {
    display: "inline-block",
  },
  breadcrumbItem: {
    display: "inline-block",
    "&:before": {
      padding: "0 5px",
      color: "#ccc",
      content: "'/'",
    },
  },
  header: {
    marginTop: "25px",
    marginBottom: "25px",
    borderBottom: "1px solid #eee",
    textAlign: "center",
  },
  innerPanel: {
    padding: "10px",
  },
})

export default useStyles
