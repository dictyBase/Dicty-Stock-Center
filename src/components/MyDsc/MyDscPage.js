// @flow
import React from "react"
import { connect } from "react-redux"
import Grid from "@material-ui/core/Grid"
import MyDscBreadcrumbs from "./MyDscBreadcrumbs"
import MyDscHeader from "./MyDscHeader"
import MyDscMainContent from "./MyDscMainContent"
import useStyles from "./myDscStyles"

type Props = {
  /** the object that contains auth data from current state */
  auth: Object,
}

/**
 * Personalized component that displays the user's login information (ID, email, name, provider)
 */

export const MyDscPage = (props: Props) => {
  const {
    auth: {
      user: { data },
      provider,
    },
  } = props
  const classes = useStyles()

  return (
    <Grid container justify="center" className={classes.container}>
      <Grid item xs={12}>
        <MyDscBreadcrumbs />
      </Grid>
      <Grid item xs={12}>
        <MyDscHeader />
      </Grid>
      <Grid item xs={8}>
        <MyDscMainContent data={data} provider={provider} />
      </Grid>
    </Grid>
  )
}

const mapStateToProps = ({ auth }) => ({ auth })

export default connect<*, *, *, *, *, *>(mapStateToProps)(MyDscPage)
