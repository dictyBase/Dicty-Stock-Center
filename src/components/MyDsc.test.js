import React from "react"
import { BrowserRouter } from "react-router-dom"
import renderer from "react-test-renderer"
import "jest-styled-components"
import { MyDsc } from "./MyDsc"

test("matching a snapshot of MyDsc", () => {
  const auth = {
    user: {
      data: {
        id: 999999,
        attributes: {
          name: "Jane Doe",
          email: "janedoe@gmail.com"
        }
      },
      provider: "google"
    }
  }

  const component = renderer.create(
    <BrowserRouter>
      <MyDsc auth={auth} />
    </BrowserRouter>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
