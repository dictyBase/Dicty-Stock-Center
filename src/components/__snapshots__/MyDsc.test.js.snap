// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matching a snapshot of MyDsc 1`] = `
.c2 h1 {
  font-size: 62px;
  margin-top: 25px;
  margin-bottom: 25px;
  border-bottom: 1px solid #eee;
  text-align: center;
}

.c2 h2 {
  font-size: 32px;
  margin-top: 25px;
  margin-bottom: 25px;
  border-bottom: 1px solid #eee;
  text-align: center;
}

.c0 {
  padding-right: 15px;
  padding-left: 15px;
  margin-right: auto;
  margin-left: auto;
}

.c1 {
  padding: 8px 15px;
  margin-bottom: 20px;
  list-style: none;
  background-color: #f5f5f5;
  border-radius: 4px;
}

.c1 li {
  display: inline-block;
}

.c1 li + li:before {
  padding: 0 5px;
  color: #ccc;
  content: "/";
}

.c1:active {
  color: #777;
}

@media (min-width:768px) {
  .c0 {
    width: 750px;
  }
}

@media (min-width:992px) {
  .c0 {
    width: 970px;
  }
}

@media (min-width:1200px) {
  .c0 {
    width: 1170px;
  }
}

<div
  className="c0"
>
  <div
    className="sc-htpNat eDtRyC sc-bdVaJa iHZvIS"
    is={null}
  >
    <div
      className="gRrvFh"
      is={null}
    >
      <ol
        className="c1"
      >
        <li>
          <a
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li>
          My DSC
        </li>
      </ol>
    </div>
    <div
      className="gRrvFh"
      is={null}
    >
      <div
        className="c2"
      >
        <h2>
          My DSC
        </h2>
      </div>
    </div>
    <div
      className="csVOHI"
      is={null}
    >
      <div
        className="gsdTnU"
        style={
          Object {
            "marginBottom": 0,
          }
        }
      >
        <div
          className="panel-body cgBkWQ"
          open={true}
          style={Object {}}
        >
          <div
            className="DRXhs"
          />
          <h1>
            <span
              aria-hidden={true}
              className="fa fa-user"
            />
             Personal Information
          </h1>
          <hr />
          <h3>
            Id: 
            999999
          </h3>
          <h3>
            Email: 
            janedoe@gmail.com
          </h3>
          <h3>
            Name: 
            Jane Doe
          </h3>
          <h3>
            Provider: 
          </h3>
          <div
            className="DRXhs"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
